homeassistant/components/conversation/default_agent.py

```diff
@@ -235,14 +235,15 @@ async def async_recognize(
         slot_lists = self._make_slot_lists()
         intent_context = self._make_intent_context(user_input)
 
-        return await self.hass.async_add_executor_job(
-            self._recognize,
-            user_input,
-            lang_intents,
-            slot_lists,
-            intent_context,
-            language,
-        )
+        async with self._lang_lock[language]:
```

Comment:
@synesthesiam I had to add this lock to avoid a dead-lock. Is hassil thread safe?